manager/orchestrator/services.go

```diff
@@ -230,5 +245,12 @@ func getRunnableSlots(s *store.MemoryStore, serviceID string) (map[uint64]slot,
 		}
 	}
 
-	return runningSlots, nil
+	deadSlots := make(map[uint64]slot)
```

Comment:
> I changed it to only do this for tasks with DesiredState > RUNNING

This doesn't help here. If a slot has 2 tasks, `t1` desiredState NEW and `t2` desiredState DEAD, it'll create a dead slot with 1 task `t2`.